### PR TITLE
fix/SearchStatesIt_failures

### PR DESCRIPTION
**Pull request by drempapis**
It seems that there is a race condition where [this call](https://github.com/elastic/elasticsearch/blob/main/qa/ccs-rolling-upgrade-remote-cluster/src/test/java/org/elasticsearch/upgrades/SearchStatesIT.java#L56) does not pick up all three nodes that [this assert](https://github.com/elastic/elasticsearch/blob/main/qa/ccs-rolling-upgrade-remote-cluster/src/test/java/org/elasticsearch/upgrades/SearchStatesIT.java#L90) is expecting.

To mitigate and test the issue, I added an "assertBusy" busy spin in the nodes lookup, waiting until all three nodes appear.

Solves #116617 and #116618

### CODE REVIEW

**Comment by drempapis** [comment]
@javanna Is this approach enough to merge the muted tests for v9.0.0 and v8.18.0?

File: qa/ccs-rolling-upgrade-remote-cluster/src/test/java/org/elasticsearch/upgrades/SearchStatesIT.java
Comment (original-brownbear) [inline qa/ccs-rolling-upgrade-remote-cluster/src/test/java/org/elasticsearch/upgrades/SearchStatesIT.java:90]: I don't think `assertBusy` will help you here? The list is just created one by a call to `getNodes(..` so it won't ever change?

Comment (drempapis) [reply qa/ccs-rolling-upgrade-remote-cluster/src/test/java/org/elasticsearch/upgrades/SearchStatesIT.java:90]: You are right. It seems environment-dependent, and applying something like the following minimizes the possibility of the assert failing by adding extra cycles to the execution. Well, I cannot think of an alternative if there are network calls in between.

```
 final List<Node>[] remoteNodes = new List<Node>[1];
        assertBusy(() -> {
            remoteNodes[0] = getNodes(restClient);
            assertThat(remoteNodes[0], hasSize(3));
        });
```

**Review by original-brownbear — APPROVED** [review]
LGTM, if CI's happy I'm happy :) thanks!

**Review by original-brownbear — COMMENTED** [review]
One second I'm a little unsure here, discussing in Slack

**Comment by drempapis** [comment]
In this pr #116577, the `boolean bwc_tests_enabled` in `elasticsearch/build.gradle` did not flip value before merging, provoking some compatibility failures, which the elastic search machine automatically reported. There is nothing to do here; just unmute the two tests.

**Pull request merged**
